timemachine/utils.py

```diff
@@ -32,3 +32,30 @@ def pairwise_transform_and_combine(xs: Iterable[A], f: Callable[[A], B], g: Call
             yield g(prev_b, b)
 
         prev_b = b
+
+
+def fair_product_2(xs: Sequence[A], ys: Sequence[B]) -> Iterable[Tuple[A, B]]:
+    """Like py:func:`itertools.product`, except iterates over arguments in an unbiased ordering.
```

Comment:
Just because it takes two arguments -- could imagine a general variadic version (like `itertools.product`) that takes an arbitrary number of arguments and is implemented in terms of the 2-argument version, but this wasn't needed here.

Reply:
Ahh, thanks! Makes sense

(had thought it might have been vestigial, if there were multiple previous versions)